### PR TITLE
Implement, integrate, and test manure in soil.phosphorus_cycling module (4/6, `Manure`)

### LINKED ISSUE #389: refactor and test manure_leach.py (soil/phosphorus_cycling) into SC_redesign directory

### PR DESCRIPTION
**Pull request by Isgotkowitz**
# Summary
This PR adds the ability in `Manure` to leach phosphorus from surface manure pools and transfer it to runoff pools and the soil profile.

# Changes
- `manure.py`
  - `daily_manure_update()`: will be the main routine used by higher level modules to update manure pools. Currently it only calls the leach and update method, but more functionality will be placed there in the next PR.
  - `_leach_and_update_phosphorus_pools()`: handles removing water-extractable phosphorus from pools and adding that phosphorus to runoff trackers, as well as calling the method that adds leached phosphorus to the soil profile.
  - `_add_infiltrated_phosphorus_to_soil()`: adds phosphorus that has infiltrated the soil profile into the top two layers of soil, with the added phosphorus distributed as dictated by the SurPhos theoretical documentation.
  - `_determine_phosphorus_leached_from_surface()`: calculates how much phosphorus is leached from surface pools, how much of that leached phosphorus infiltrates the soil profile and how much is removed from the field by runoff.
  - All of the above methods are tested individually in `test_manure.py`. NOTE: please carefully scrutinize the code coverage of the test cases for `_determine_phosphorus_leached_from_surface()`. It is a fairly complex method that has to ensure no more phosphorus can be removed from the field than there actually is on the field.
- `soil_data.py`
  - now separately tracks the phosphorus in runoff from each individual pool, and resets the trackers annually with all other annual tracker attributes in `SoilData`
  - `test_soil_data_and_config_factory.py` now tests that the new annual tracker attributes are reset correctly

# Notes 
This PR was originally meant to be the penultimate one in the series to implement `Manure`, but an extra PR will most likely be needed to implement the decomposition functionality of `Manure`.

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py [hunk 1]
```diff
@@ -24,7 +25,159 @@ def __init__(self, soil_data: Optional[SoilData] = None):
...
 
+    def daily_manure_update(self, rainfall: float, runoff: float, field_size: float) -> None:
+        """This method conducts daily operations on manure including decomposition, assimilation into soil, etc.
+
+        Parameters
+        ----------
+        rainfall : float
+            The amount of rainfall on the current day (mm)
+        runoff : float
+            The amount of runoff on the current day (mm)
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py]:
```suggestion
            The amount of rainfall runoff on the current day (mm)
```

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py [hunk 2]
```diff
@@ -24,7 +25,159 @@ def __init__(self, soil_data: Optional[SoilData] = None):
...
+                                                   water_extractable_phosphorus: float, is_organic: bool) -> Dict:
+        """This method determines how much phosphorus is leached from the given pool, how that phosphorus is distributed
+            between runoff and soil infiltration, and how much phosphorus remains in the given pool.
+
+        Parameters
+        ----------
+        rainfall : float
+            The amount of rainfall on the current day (mm)
+        runoff : float
+            The amount of runoff on the current day (mm)
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py]: same as above

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py [hunk 3]
```diff
@@ -24,7 +25,159 @@ def __init__(self, soil_data: Optional[SoilData] = None):
...
+            Area of the field (ha)
+        manure_dry_mass : float
+            Dry-weight equivalent of manure on the field (kg)
+        field_coverage : float
+            Percent of the field covered by manure, in range [0.0, 1.0]
+        water_extractable_phosphorus : float
+            The mass of the water extractable phosphorus pool that is being leached from (kg)
+        is_organic : bool
+            Is the phosphorus being leached organic (True / False)
+
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py:153]: Any reference to these calculations? It's better to add them if there are.

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py [hunk 4]
```diff
@@ -24,7 +25,159 @@ def __init__(self, soil_data: Optional[SoilData] = None):
...
+        runoff : float
+            The amount of runoff on the current day (mm)
+        field_size : float
+            The size of the field (ha)
+
+        """
+        if rainfall > 0:
+            self._leach_and_update_phosphorus_pools(rainfall, runoff, field_size)
+
+    def _leach_and_update_phosphorus_pools(self, rainfall: float, runoff: float, field_size: float) -> None:
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py:44]: perhaps a more detailed name for runoff? There are quite some different kinds of runoff in this module.

File: SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py [hunk 5]
```diff
@@ -24,7 +25,159 @@ def __init__(self, soil_data: Optional[SoilData] = None):
...
+        Parameters
+        ----------
+        infiltrated_phosphorus_amount : float
+            The amount of phosphorus to be added to the soil profile (kg)
+        field_size : float
+            The size of the field (ha)
+
+        Notes
+        -----
+        This method follows what is outlined in SurPhos (theoretical documentation, page 8, paragraph just below eqn.
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/soil/phosphorus_cycling/manure.py:119]: Maybe also put this in reference?

**Review by matthew7838 — APPROVED** [review]
Looks good overall, just have a few suggestions on naming and descriptions. Nice PR!

**Review by morrowcj — APPROVED** [review]
LGTM

**Pull request merged**